transport_plugins/bled112/iotile_transport_bled112/bled112.py

```diff
@@ -504,15 +505,142 @@ def _handle_event(self, event):
         else:
             self._logger.warn('Unhandled BLE event: ' + str(event))
 
+
     def _parse_scan_response(self, response):
         """Parse the IOTile specific data structures in the BLE advertisement packets and add the device to our list of scanned devices
         """
+        self._logger.debug("   call: _parse_scan_response({0})".format(response))
+        # Parse advertisement and determine if it matches V1 or V2.
 
         payload = response.payload
         length = len(payload) - 10
 
         if length < 0:
-            return  # FIXME: Log an error here
+            self._logger.error("BLE Advertising packet payload is empty.")
```

Comment:
I actually think that old `FIXME` is wrong and we should not log an error here.  This will log an error for all non-iotile devices that don't happen to have a lot of scan response data.